astropy/wcs/src/wcslib_wrap.c

```diff
@@ -434,6 +435,7 @@ PyWcsprm_init(
       return -1;
     }
 
+    self->x.flag = -1;
```

Comment:
I think that's wrong, but I may be mistaken. `self->x` should already have allocated memory, so having a -1 would lead to a memory leak.

If this is necessary for `wcscopy` (I haven't seen something like that in the [documentation](https://www.atnf.csiro.au/people/mcalabre/WCS/wcslib/wcs_8h.html#ac55946dadc53ac592cb686275902ae7b)) then we probably have to adjust a few more calls we have in our extension and it would be good to ask for a clarification on this in the documentation.

Or am I missing something?

Reply:
It could be problematic because `__init__` may be called multiple times on the same wcs object and then the flag shouldn't be set to `-1` for subsequent calls. I'm not sure if that's possible through our `WCS` wrapper.